cmd/crates/soroban-test/tests/it/integration/util.rs

```diff
@@ -63,15 +63,6 @@ pub async fn deploy_custom_account(sandbox: &TestEnv) -> String {
     deploy_contract(sandbox, CUSTOM_ACCOUNT, DeployOptions::default()).await
 }
 
-pub fn setup_env_for_sandbox(sandbox: &TestEnv) {
-    env::set_var("SOROBAN_ACCOUNT", TEST_ACCOUNT);
-    env::set_var("SOROBAN_RPC_URL", sandbox.network.rpc_url.clone());
-    env::set_var("SOROBAN_NETWORK_PASSPHRASE", LOCAL_NETWORK_PASSPHRASE);
-
-    env::set_var("XDG_CONFIG_HOME", sandbox.config_dir().as_os_str());
-    env::set_var("XDG_DATA_HOME", sandbox.data_dir().as_os_str());
-}
```

Comment:
These lines globber the environment and affect the state of the entire process, every test in it. That's not ideal because it means one test has a side-effect on the behaviour of the rest of the tests. It also presents us from using `stellar keys use ...` in cookbook tests.

Most of these lines aren't required anyway because we set the same parameters other ways elsewhere. The couple that are required, the config dir, can be set via the `--config-dir` option instead. The `setup_env_for_sandbox` function is only used in one place, in the `deploy_contract` function, so I've moved setting the config-dir to there.